RecoTauTag/HLTProducers/src/HLTTauIPFilter.cc

```diff
@@ -0,0 +1,94 @@
+#include "FWCore/Utilities/interface/Exception.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "DataFormats/TauReco/interface/PFTau.h"
+#include "DataFormats/TauReco/interface/PFTauFwd.h"
+#include "DataFormats/TauReco/interface/PFTauTransverseImpactParameterAssociation.h"
+#include "HLTrigger/HLTcore/interface/HLTFilter.h"
+#include "CommonTools/Utils/interface/StringCutObjectSelector.h"
+#include <vector>
+
+class HLTTauIPFilter : public HLTFilter {
+public:
+  explicit HLTTauIPFilter(const edm::ParameterSet& config);
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  bool hltFilter(edm::Event& event,
+                 const edm::EventSetup& setup,
+                 trigger::TriggerFilterObjectWithRefs& filterproduct) const override;
+
+private:
+  const edm::InputTag m_TausInputTag;
+  const edm::EDGetTokenT<reco::PFTauCollection> m_TausToken;
+  const edm::EDGetTokenT<edm::AssociationVector<reco::PFTauRefProd, std::vector<reco::PFTauTransverseImpactParameterRef>>>
+      m_TausTIPToken;
+
+  const int m_MinTaus;  // min. number of taus required to pass the cuts
+  std::unique_ptr<StringCutObjectSelector<reco::PFTauTransverseImpactParameter>> m_cut;
+  std::string m_cutString;
+  const int m_TriggerType;
+};
+
+HLTTauIPFilter::HLTTauIPFilter(const edm::ParameterSet& config)
+    : HLTFilter(config),
+      m_TausInputTag(config.getParameter<edm::InputTag>("Taus")),
+      m_TausToken(consumes(m_TausInputTag)),
+      m_TausTIPToken(consumes(config.getParameter<edm::InputTag>("TausIP"))),
+      m_MinTaus(config.getParameter<int>("MinN")),
+      m_cut(std::make_unique<StringCutObjectSelector<reco::PFTauTransverseImpactParameter>>(
+          config.getParameter<std::string>("Cut"))),
+      m_cutString(config.getParameter<std::string>("Cut")),
+      m_TriggerType(config.getParameter<int>("TriggerType")) {
+  edm::LogInfo("") << " (HLTTauIPFilter) trigger cuts:\n"
+                   << " cut string value: [" << m_cutString << "]\n"
+                   << " min no. passing taus: " << m_MinTaus << ", TriggerType: " << m_TriggerType;
+}
+
+void HLTTauIPFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  makeHLTFilterDescription(desc);
+  desc.add<edm::InputTag>("Taus", edm::InputTag("hltTauCollection"));
+  desc.add<edm::InputTag>("TausIP", edm::InputTag("hltTauIPCollection"));
+  desc.add<int>("MinN", 1);
+  desc.add<int>("TriggerType", 84);
+  desc.add<std::string>("Cut", "dxy > -999. ");
+  descriptions.addWithDefaultLabel(desc);
+}
+
+bool HLTTauIPFilter::hltFilter(edm::Event& event,
+                               const edm::EventSetup& setup,
+                               trigger::TriggerFilterObjectWithRefs& filterproduct) const {
+  auto const m_Taus = event.getHandle(m_TausToken);
+  if (saveTags())
+    filterproduct.addCollectionTag(m_TausInputTag);
+
+  auto const& m_TausTIP = event.get(m_TausTIPToken);
```

Comment:
```suggestion
  auto const h_Taus = event.getHandle(m_TausToken);
  if (saveTags())
    filterproduct.addCollectionTag(m_TausInputTag);

  auto const& tausTIP = event.get(m_TausTIPToken);
```
In this plugin, `m_*` names are used for class members, so shouldn't be used for 'local' objects too (`h_` can help identifying that obj as a Handle).